### PR TITLE
[BUGFIX] Remove test set from validation

### LINKED ISSUE #372: training scripts should not evaluate test set
Currently, the MT training scripts do training (with dev set) and evaluation (on the test set) altogether.
Ideally, the training script should not touch test set because reporting numbers on the test set (even if it's not used for model selection) is already leaking test set information.

For example, in [`fairseq`](https://github.com/pytorch/fairseq), train/validation (train.py) and evaluate (generate.py) are in two separate files.

### PR DESCRIPTION
**Pull request by avinashsai**
Fixes #372

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1228/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1228/1/index.html

**Comment by avinashsai** [comment]
@leezu @szha Please review this

**Comment by avinashsai** [comment]
Similarly, an argument should be added in ```train_transformer.py``` whether to perform inference on test set? It should be default set to ```true```?

**Comment by avinashsai** [comment]
@StrayBird-ATSH GluonNLP-py3-master-gpu-doc is failing.

**Comment by StrayBird-ATSH** [comment]
Thanks for reminding. I have noticed it on some other pull request. It might be due to some global internet connectivity failures. We are sorting out reasons.

**Comment by avinashsai** [comment]
@StrayBird-ATSH Any updated on this?

**Comment by avinashsai** [comment]
@StrayBird-ATSH can you restart builds to see if the issue has been resolved?

**Comment by StrayBird-ATSH** [comment]
OK. I will have a look into it.

**Comment by avinashsai** [comment]
@StrayBird-ATSH I guess the error is fixed. If possible, can you restart builds?

**Comment by StrayBird-ATSH** [comment]
> @StrayBird-ATSH I guess the error is fixed. If possible, can you restart builds?

Yes, would you please merge the latest `master` branch into your pull request, especially to include #1236 ?
Thanks!

**Comment by avinashsai** [comment]
@StrayBird-ATSH There are other errors in the same build. Looks like, few more might be identified.

**Comment by mli** [comment]
Job PR-1228/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1228/10/index.html

**Comment by avinashsai** [comment]
@eric-haibin-lin please review this

**Comment by eric-haibin-lin** [comment]
thanks for the contribution!
